### ISSUE TITLE
[IOS] Validate receipt arrive to server with id = BUNDLE-ID of my App but not with id of my real product in Apple

### ISSUE THREAD

**Issue by r-rodriguez-s** [j3k0/cordova-plugin-purchase#1156]
Im using last Xcode (CATALINA) and using WkWebview Cordova IOS@6.1.1
Plugin version cc.fovea.cordova.purchase@10.1.1
Device: iPhone iOS 14.3

### system info
When a user buy a suscription product the receipt validation we receive in the server do not have transaction[id]
and id of the product allways come with id of the App Bundle but not del id of the suscription in Apple. 

But allways transaction[appStoreReceipt] is correct and wen we call Apple receipt with it sometimes Apple receipt "in_app": [] is void.

¿Is this a problem of the plugin or something with Apple or is this plugin incompatible with WkWebview? 

Please, we are very worried no because we think we are losing clients. Any help is very apreciated.

This is an example of the vaidate receipt the server get form the App

{
    "id": "com.facetoned.app THIS IS TE ID BUNDLE OF THE APP NOT A PRODUCT",
    "alias": "application",
    "type": "application",
    "group": "",
    "state": "approved",
    "title": "Application Bundle",
    "description": null,
    "priceMicros": null,
    "price": null,
    "currency": null,
    "countryCode": null,
    "loaded": true,
    "canPurchase": false,
    "owned": false,
    "introPrice": null,
    "introPriceMicros": null,
    "introPricePeriod": null,
    "introPriceNumberOfPeriods": null,
    "introPricePeriodUnit": null,
    "introPriceSubscriptionPeriod": null,
    "introPricePaymentMode": null,
    "ineligibleForIntroPrice": null,
    "discounts": [],
    "downloading": false,
    "downloaded": false,
    "additionalData": [],
    "transaction": {
        "type": "ios-appstore",
        "appStoreReceipt": "LONG B64 FROM APPLE"
    },
    "version": "1.2.0",
    "valid": true,
    "device": {
        "plugin": "cordova-plugin-purchase\/10.1.0",
        "cordova": "6.1.1",
        "model": "iPhone11,6",
        "platform": "iOS",
        "version": "14.3",
        "manufacturer": "Apple",
        "fingerprint": "729a5f4f156d8c0c70c2759b44144c79"
    }
}

Sorry my English.

**Comment by OrthoCube** [comment]
Apparently, this is an actual feature that is used by the plugin to check if the application is legitimately downloaded from the App Store, so it validates a receipt of type=application to verify this. More details can be found here: https://github.com/j3k0/cordova-plugin-purchase/issues/1155

**Comment by r-rodriguez-s** [comment]
Thank you very much for your answer. 

It so happened that today Apple published a new version of our App and We see this type of Apple Validates Receipt in the logs and We were scared thinking that it could be something of the compatibility between this plugin and WkWebview or something like that.


Thank you very much.

**Comment by OrthoCube** [comment]
Yeah, as far as I know, this is not written in the documentation, so even I did not know what this is. But it seems to be the correct function, so as long as you are using a validation service that responds correctly to receipts that have type=application (for example, Fovea, or maybe your own validation server), then there should be no problem, I think.

We were using our own validation server which did not respond correctly to type=application receipts, so that caused problems, but after fixing it, it did work, I think.

**Comment by r-rodriguez-s** [comment]
Currently what we do is send to Apple the string **B64** contained in the **appStoreReceipt** property of the transaction section 

Apple returns the value of **"in_app": []** as empty. And then we send to the App an **ok = false** a  validate.


Do you think that I should still send an **ok = true** as validation and then app will go better?


I thank you again for all this help from you

**Comment by OrthoCube** [comment]
Hmm. I'm not sure about that, but even though there is no purchase yet, as long as it's a valid B64 string, then I think your validator should still reply with ok=true. In my observation, if you reply with ok=false, the plugin stops working completely.

**Comment by j3k0** [comment]
`in_app` being empty doesn't mean the receipt is not valid. If you reply `ok: false` it means the app receipt is not legit, which the plugin considers a fatal error.

Note that this also allows you to check things like the first version of the app that the user downloaded, which might be useful if you turn a free feature into a paid feature (but want to keep it free for users that first downloaded an older version).

**Comment by r-rodriguez-s** [comment]
Just in case this helps someone.

According to this thread, my server was replying to validate with ok = true as soon as it received a validate receipt arrive to server with id = BUNDLE-ID and without a transaction id

I recently began to receive complaints from users who had paid and had not registered them on my server.
Now I realized that whenever a validate receipt arrive to server, whenever it contains a base64 in the appStoreReceipt, I must check with Apple because it can be a purchase and until now I did not consider them.

That is to say, the important thing is to always check against apple the base64 of the appStoreReceipt although in the JSON received from the app in the server during the validate some info were missing (sometimes, not always).
I don't know why sometimes yes, sometimes not the JSON validate in the first buy form a user do not arrive to server with product name or transactionId, but always check the receipt with apple an if it is correct then we consider the user buy is ok.

Thank you for your help
